cranelift/codegen/meta/src/gen_listing.rs

```diff
@@ -0,0 +1,171 @@
+//! Generate human-readable lists of instructions and their encodings.
+//!
+//! To display the generated contents, one could use
+//! `find target -name 'listing-x86*' | xargs cat | less` from the top-level project directory.
+use std::collections::BTreeMap;
+use std::fmt::{Debug, Display, Error, Formatter as DisplayFormatter};
+
+use crate::cdsl::encodings::EncodingContent;
+use crate::cdsl::isa::TargetIsa;
+use crate::cdsl::recipes::{EncodingRecipe, EncodingRecipeNumber, OperandConstraint};
+use crate::cdsl::regs::{IsaRegs, RegClass, RegClassIndex};
+use crate::error;
+use crate::shared::Definitions as SharedDefinitions;
+use crate::srcgen::Formatter;
+
+/// Emit a list of all encodings in a human-readable format.
+fn emit_listings(_: &SharedDefinitions, isa: &TargetIsa, fmt: &mut Formatter) {
+    let mut recipe_counter = BTreeMap::new();
```

Comment:
Do we rely on remembering the order of insertions? If not, we could use a plain HashMap here.